napalm_panos/panos.py

```diff
@@ -0,0 +1,326 @@
+# Copyright 2015 Spotify AB. All rights reserved.
+#
+# The contents of this file are licensed under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# std libs
+import sys
+import xmltodict
+import json
+import pan.xapi
+import os.path
+import xml.etree
+import os
+import requests
+import requests_toolbelt
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+from datetime import datetime
+import time
+
+# local modules
+import napalm_base.exceptions
+import napalm_base.helpers
+from napalm_base.exceptions import ReplaceConfigException, MergeConfigException
+
+from napalm_base.base import NetworkDriver
+from netmiko import ConnectHandler
+
+
+class PANOSDriver(NetworkDriver):
+
+    def __init__(self, hostname, username, password, timeout=60):
+        self.hostname = hostname
+        self.username = username
+        self.password = password
+        self.timeout = timeout
+        self.loaded = False
+        self.changed = False
+        self.device = None
+        self.connected = False
+
+    def __enter__(self):
```

Comment:
As @mirceaulinic pointed out, do not override methods that have code in `napalm-base` unless it's super strictly necessary. And when I mean super strictly necessary I really mean it ;)